types/model/name.go

```diff
@@ -571,6 +572,87 @@ func (r Name) IsValid() bool {
 	return r.parts[PartModel] != ""
 }
 
+// ParseNameFromPath parses forms of a URL path into a Name. Specifically,
+// it trims any leading "/" and then calls [ParseName] with fill.
+func ParseNameFromPath(s, fill string) Name {
+	s = strings.TrimPrefix(s, "/")
+	return ParseName(s, fill)
+}
+
+// Path returns a complete, canonicalized, relative URL path using the parts of a
+// complete Name.
+//
+// The parts maintain their original case.
+//
+// It panics if r is incomplete. Callers should check [Name.IsComplete] before
+// calling this method.
+//
+// Example:
+//
+//	ParseName("example.com/namespace/model:tag+build").Path() // returns "/example.com/namespace/model:tag+build"
+func (r Name) Path() string {
```

Comment:
This is very easily confused with `Filepath`. Maybe rename this to `URL()` and return `*url.URL`?

Reply:
Strings are more ergonomic and useful when working with the `path` package. Returning a `*url.URL` suggests Name came from or is a URL, and it technically is not. It could easily be a path too. 

I do agree that path can be conflated with filepath, so I renamed to `URLPath`. I think that's a fair compromise.